library/Icingadb/Web/Control/SearchBar/ObjectSuggestions.php

```diff
@@ -221,41 +221,98 @@ protected function fetchColumnSuggestions($searchTerm)
         $model = $this->getModel();
         $query = $model::on($this->getDb());
 
-        // Ordinary columns first
-        foreach (self::collectFilterColumns($model, $query->getResolver()) as $columnName => $columnMeta) {
-            yield $columnName => $columnMeta;
-        }
-
-        // Custom variables only after the columns are exhausted and there's actually a chance the user sees them
-        $titleAdded = false;
+        // Collect custom variables first and check for exact matches
         $parsedArrayVars = [];
-        foreach ($this->getDb()->select($this->queryCustomvarConfig($searchTerm)) as $customVar) {
-            $search = $name = $customVar->flatname;
-            if (preg_match('/\w+(?:\[(\d*)])+$/', $search, $matches)) {
-                $name = substr($search, 0, -(strlen($matches[1]) + 2));
-                if (isset($parsedArrayVars[$name])) {
-                    continue;
+        $varSuggestions = [];
+        $exactVarMatches = [];
+        $halfLimit = (int) (static::DEFAULT_LIMIT / 2);
+
+        $exactSearchTerm = trim($searchTerm, ' *');
+        if ($exactSearchTerm !== '') {
+            foreach ($this->getDb()->select($this->queryCustomvarConfig($searchTerm)) as $customVar) {
+                $search = $name = $customVar->flatname;
+                if (preg_match('/\w+(?:\[(\d*)])+$/', $search, $matches)) {
+                    $name = substr($search, 0, -(strlen($matches[1]) + 2));
+                    if (isset($parsedArrayVars[$name])) {
+                        continue;
+                    }
+
+                    $parsedArrayVars[$name] = true;
+                    $search = $name . '[*]';
                 }
 
-                $parsedArrayVars[$name] = true;
-                $search = $name . '[*]';
+                foreach ($this->customVarSources as $relation => $label) {
+                    if (isset($customVar->$relation)) {
+                        $varRelation = $relation . '.vars.' . $search;
+                        $varLabel = sprintf($label, $name);
+                        if ($search === $exactSearchTerm) {
```

Comment:
`$search` is changed in case of arrays and such won't appear as best suggestion as a result. Use `$name` instead.